src/PharoLauncher-CLI/PhLProcessWrapper.extension.st

```diff
@@ -0,0 +1,33 @@
+Extension { #name : #PhLProcessWrapper }
+
+{ #category : #'*PharoLauncher-CLI' }
+PhLProcessWrapper >> getExceptionMessageFor: cmdString status: statusMsg stdError: errorString [
+
+	^ String streamContents: [:aStream | 
+		aStream 
+			nextPutAll: ('OS process command: ''{1}''  exited with: {2}.' format: {cmdString. statusMsg.});
+			nextPutAll: OSPlatform current lineEnding;
+			nextPutAll: ('Stderr contents: "{1}".' format: {errorString}).
+	]
+]
+
+{ #category : #'*PharoLauncher-CLI' }
+PhLProcessWrapper >> runAndWaitWithStdOutput [
+	"this is helper method to catch std. output and return to sender of this method, 
+	 exit codes are handled and proper exception is raised, when not successful
+	"
+
+	self prepareProcess.
+	process
+		redirectStdout;
+		redirectStderr;
+		runAndWaitOnExitDo: [ :termProcess :outString :errString | 
```

Comment:
All of this is specific to Unix (methods defined on OSSSUnixSubprocess.
PhLProcessWrapper is a common API for both Unix and Windows (OSSubprocess and OSWindSubProcess) and should work on both OS.
I suggest to move the method content in a linux specific method like #runOnLinuxAndWaitWithStdOutput. #runAndWaitWithStdOutput will do a dispatch to this method (like #runAndWaitTimeOut:) and you should define a method #runOnWindowsAndWaitWithStdOutput that defaults to a runAndWait for now (std outputs not yet managed in OsWinSubProcess)

Reply:
Got it, will change that according to description!

Reply:
@demarey : I've moved implementation of this medthod to OSUnixSubprocess and OSWinSubprocess classes, in order to not make dispatch in "ugly" way. Do you agree? There is following:
```
PhLProcessWrapper>>runAndWaitWithStdOutput
	"this is helper method to catch std. output and return to sender of this method, 
	 exit codes are handled and proper exception is raised, when not successful
	"

	self prepareProcess.
	"return content of stdout, may throw exception that can be handled by sender of this method"
	^ process runAndWaitWithStdOutput 
```
and then: 
```
OSUnixSubprocess>>runAndWaitWithStdOutput
	self 
		redirectStdout;
		redirectStderr;
		runAndWaitOnExitDo: [ .....]
```
and Windows: 
```
OSUnixSubprocess>>runAndWaitWithStdOutput
      self runAndWait.
      "string value is expected"
      ^ ''
```

Reply:
@Bajger. It is ok for me if they are packaged as extensions and not directly in OSUnixSubprocess and OSWinSubprocess projects

Reply:
moved implementation to OSUnixSubprocess and OSWinProcess as extension methods